library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -13,7 +13,13 @@ Http::FilterHeadersStatus NetworkConfigurationFilter::decodeHeaders(Http::Reques
   envoy_network_t network = network_configurator_->getPreferredNetwork();
   ENVOY_LOG(debug, "current preferred network: {}", network);
 
-  auto connection_options = network_configurator_->getUpstreamSocketOptions(network);
+  if (enable_interface_binding_) {
+    override_interface_ = network_configurator_->overrideInterface(network);
+  }
+  ENVOY_LOG(debug, "will override interface: {}", override_interface_);
```

Comment:
Should we use `ENVOY_EVENT_LOG` here to get more visibility into this feature? If not, what's our plan to get instrumentation for interface binding?

Reply:
Should we include the information about the current/target interfaces in here (if possible)?

Reply:
I'm planning to add event logs last, once I'm sure of the shape of things.